### PR TITLE
Script Creation in back-end

### PR DESCRIPTION
**Pull request by bvobart**
It is now possible to create scripts in the back-end. JUnit tests and a simple demo application have been provided in the tests folder.

### CODE REVIEW

File: src/main/java/nl/tudelft/contextproject/script/Script.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
+package main.java.nl.tudelft.contextproject.script;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+
+
```
Comment (stevenmeijer) [inline src/main/java/nl/tudelft/contextproject/script/Script.java]: Double vertical whitespaces are discouraged

File: src/test/java/nl/tudelft/contextproject/camera/CameraTest.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        cam.addObserver(testOb);
+        cam.focus(-65);
+        assertEquals(1365, cam.getSettings().getFocus());
+        assertTrue(testOb.wasCalled());
+        cam.focus(656565);
+        assertEquals(CameraSettings.FOCUS_LIMIT_HIGH, cam.getSettings().getFocus());
+    }
+
+
+
```
Comment (stevenmeijer) [inline src/test/java/nl/tudelft/contextproject/camera/CameraTest.java]: Triple vertical whitespaces are a bad coding style

File: src/main/java/nl/tudelft/contextproject/presets/Preset.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    private String name;
+    private CameraSettings toSet;
+    
+    /**
+     * Creates a Preset object with to set camera settings toSet.
+     * 
+     * @param toSet Camera settings to set when applied.
+     */
+    protected Preset(CameraSettings toSet) {
+        name = "";
```
Comment (stevenmeijer) [inline src/main/java/nl/tudelft/contextproject/presets/Preset.java:28]: Naming should consistently use "this."

File: src/main/java/nl/tudelft/contextproject/script/Shot.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+     */
+    public Shot(int num, Camera cam, Preset pres) {
+        this.number = num;
+        this.camera = cam;
+        this.preset = pres;
+    }
+
+    /**
+     * Returns the number of the shot.
+     * @return the number of the shot.
```
Comment (stevenmeijer) [inline src/main/java/nl/tudelft/contextproject/script/Shot.java:33]: This javadoc is incorrectly formulated (same goes for the rest of the class). See https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment

File: src/main/java/nl/tudelft/contextproject/presets/Preset.java (hunk 3 above)
Comment (bvobart) [inline src/main/java/nl/tudelft/contextproject/presets/Preset.java]: @TH3steven I've fixed the whitespace issues, but do we really need to use `this`? Personally, I use it only where necessary due to shadowing.

Comment (bvobart) [inline src/main/java/nl/tudelft/contextproject/presets/Preset.java:28]: @TH3steven I've fixed the whitespace issues, but do we really need to use `this.`? Personally, I use it only where necessary due to shadowing.

Comment (stevenmeijer) [inline src/main/java/nl/tudelft/contextproject/presets/Preset.java:28]: Nope not necessary, my bad

File: src/main/java/nl/tudelft/contextproject/script/Shot.java (hunk 4 above)
Comment (bvobart) [inline src/main/java/nl/tudelft/contextproject/script/Shot.java:33]: @TH3steven Only the Javadoc of the `execute()` method is not correctly formulated according to the style guide. I've fixed this in commit f47ea860e8e14874a232a8ce6c3b418b4b4decad

**Comment by tkolenbrander** [comment]
After fixing the checkstyle issues, I approve it!

**Pull request merged**